Automattic/wp-calypso
# Activity Log: Restore `react-intersection-observer` usage

See https://github.com/Automattic/wp-calypso/issues/74022
Previously https://github.com/Automattic/wp-calypso/pull/69119

## Proposed Changes

Restores use of `react-intersection-observer` in the Activity Log because `calypso/lib/use-in-view` is buggy and it doesn't make sense to spend time reinventing the wheel.

## Testing Instructions

1. Apply the following patch to have the banner appear:

```diff
diff --git a/client/components/activity-card-list/visible-days-limit-upsell/index.tsx b/client/components/activity-card-list/visible-days-limit-upsell/index.tsx
index 03c6b5639a..9712dccd52 100644
--- a/client/components/activity-card-list/visible-days-limit-upsell/index.tsx
+++ b/client/components/activity-card-list/visible-days-limit-upsell/index.tsx
@@ -56,10 +56,6 @@ const VisibleDaysLimitUpsell: React.FC< OwnProps > = ( { cardClassName } ) => {
 
 	const upsellRef = useTrackUpsellView( siteId );
 
-	if ( ! Number.isInteger( visibleDays ) ) {
-		return null;
-	}
-
 	const card =
 		isJetpackCloud() || isEnabled( 'activity-log/v2' ) ? (
 			<ActivityCard className={ cardClassName } activity={ PLACEHOLDER_ACTIVITY } />
diff --git a/client/my-sites/activity/activity-log/index.jsx b/client/my-sites/activity/activity-log/index.jsx
index 2bad1c2a88..5134631420 100644
--- a/client/my-sites/activity/activity-log/index.jsx
+++ b/client/my-sites/activity/activity-log/index.jsx
@@ -547,9 +547,7 @@ class ActivityLog extends Component {
 								)
 							) }
 						</section>
-						{ showVisibleDaysLimitUpsell && (
-							<VisibleDaysLimitUpsell cardClassName="activity-log-item__card" />
-						) }
+						<VisibleDaysLimitUpsell cardClassName="activity-log-item__card" />
 						{ ! hasFullActivityLog && ! isIntroDismissed && <UpgradeBanner siteId={ siteId } /> }
 						<Pagination
 							compact={ isMobile() }

```

<img width="1089" alt="image" src="https://user-images.githubusercontent.com/36432/231005081-fa0deab7-d771-4788-8617-90fe7c698700.png">

2. On a WoA Business site, navigate to 'Activity Log'
3. Scroll to the bottom of the page and verify the `calypso_activitylog_visiblelimit_upsell_view` event only appears once.
